### PR TITLE
Krishna/2225/remove timestamp from block header

### LINKED ISSUE #2225: Remove timestamp field from block header
## Description
The block header contains a `Timestamp` field, which is set, but never read.

It’s problematic to use it anywhere since it’s un-vetted, one can plaster any timestamp that serves them onto anything.
Since we’re not doing anything with this field we may as well call it “arbitrary 8 bytes that the miner can fill with whatever they please”.

## Affected code
https://github.com/spacemeshos/go-spacemesh/blob/f76c8b99427a4a9f559da87fced6cf5aa3219266/common/types/block.go#L124-L133

### PR DESCRIPTION
**Pull request by krishnaindani**
## Motivation

Closes #2225

## Changes
Remove Timestamp field from the BlockHeader struct 

## TODO

- [ ] Explain motivation or link existing issue(s)(https://github.com/spacemeshos/go-spacemesh/issues/2225)
- [ ] Test changes and document test plan
- [ ] Update documentation as needed

### CODE REVIEW

File: .gitignore
Comment (noamnelke) [inline .gitignore]: Do you mind adding a newline at the end?

**Review by noamnelke — APPROVED** [review]
LGTM

File: .gitignore
Comment (krishnaindani) [reply .gitignore]: sure. Just added

**Comment by krishnaindani** [comment]
> LGTM

Thank you for the review.

**Comment by krishnaindani** [comment]
@noamnelke how do the merge process looks like? Will you or one of the maintainers merge it, because I don't see I have permission to merge?

**Comment by noamnelke** [comment]
> Will you or one of the maintainers merge it

Yes, we'll merge it. We currently have some stability issues with our system tests, which are a requirement for a merge, so I was considering waiting for those to be resolved before merging, but I'll just try now (it may take several attempts until the issue is solved).

**Comment by noamnelke** [comment]
bors merge

**Comment by noamnelke** [comment]
bors merge

**Comment by krishnaindani** [comment]
> > Will you or one of the maintainers merge it
> 
> Yes, we'll merge it. We currently have some stability issues with our system tests, which are a requirement for a merge, so I was considering waiting for those to be resolved before merging, but I'll just try now (it may take several attempts until the issue is solved).

not a problem. Just wanted to know. Appreciate for all the help.

**Comment by noamnelke** [comment]
bors merge

**Comment by noamnelke** [comment]
bors merge

**Comment by noamnelke** [comment]
bors merge

**Comment by noamnelke** [comment]
bors merge

**Comment by noamnelke** [comment]
@krishnaindani I found an issue in the last failed test run. There's a test that was only passing because of the `Timestamp` field. The test is incorrect and needs to be patched.

Can you apply the following patch?

```patch
diff --git a/miner/builder_test.go b/miner/builder_test.go
index 2fa37242..65a1a414 100644
--- a/miner/builder_test.go
+++ b/miner/builder_test.go
@@ -183,12 +183,14 @@ func TestBlockBuilder_BlockIdGeneration(t *testing.T) {
 	builder2.AtxDb = atxDbMock{}
 	builder1.AtxDb = atxDbMock{}
 
-	b1, err := builder1.createBlock(types.GetEffectiveGenesis()+2, types.ATXID{}, types.BlockEligibilityProof{}, nil)
+	atxId1 := types.ATXID(types.HexToHash32("dead"))
+	atxId2 := types.ATXID(types.HexToHash32("beef"))
+	b1, err := builder1.createBlock(types.GetEffectiveGenesis()+2, atxId1, types.BlockEligibilityProof{}, nil)
 	assert.NoError(t, err)
-	b2, err := builder2.createBlock(types.GetEffectiveGenesis()+2, types.ATXID{}, types.BlockEligibilityProof{}, nil)
+	b2, err := builder2.createBlock(types.GetEffectiveGenesis()+2, atxId2, types.BlockEligibilityProof{}, nil)
 	assert.NoError(t, err)
 
-	assert.True(t, b1.ID() != b2.ID(), "ids are identical")
+	assert.NotEqual(t, b1.ID(), b2.ID(), "ids are identical")
 }
 
 var (
```

**Comment by krishnaindani** [comment]
> @krishnaindani I found an issue in the last failed test run. There's a test that was only passing because of the `Timestamp` field. The test is incorrect and needs to be patched.
> 
> Can you apply the following patch?
> 
> ```diff
> diff --git a/miner/builder_test.go b/miner/builder_test.go
> index 2fa37242..65a1a414 100644
> --- a/miner/builder_test.go
> +++ b/miner/builder_test.go
> @@ -183,12 +183,14 @@ func TestBlockBuilder_BlockIdGeneration(t *testing.T) {
>  	builder2.AtxDb = atxDbMock{}
>  	builder1.AtxDb = atxDbMock{}
>  
> -	b1, err := builder1.createBlock(types.GetEffectiveGenesis()+2, types.ATXID{}, types.BlockEligibilityProof{}, nil)
> +	atxId1 := types.ATXID(types.HexToHash32("dead"))
> +	atxId2 := types.ATXID(types.HexToHash32("beef"))
> +	b1, err := builder1.createBlock(types.GetEffectiveGenesis()+2, atxId1, types.BlockEligibilityProof{}, nil)
>  	assert.NoError(t, err)
> -	b2, err := builder2.createBlock(types.GetEffectiveGenesis()+2, types.ATXID{}, types.BlockEligibilityProof{}, nil)
> +	b2, err := builder2.createBlock(types.GetEffectiveGenesis()+2, atxId2, types.BlockEligibilityProof{}, nil)
>  	assert.NoError(t, err)
>  
> -	assert.True(t, b1.ID() != b2.ID(), "ids are identical")
> +	assert.NotEqual(t, b1.ID(), b2.ID(), "ids are identical")
>  }
>  
>  var (
> ```

sure. Let me take a look.

**Comment by antonlerner** [comment]
closed due to inactivity

**Pull request closed without merging**
